### DISCUSSION TITLE
Apply labels automatically at issue creation

### DISCUSSION THREAD

**Discussion by alalazo** [community/community#23682] [New to GitHub]
<p>I have seen the documentation on how to write <a href="https://help.github.com/articles/about-issue-and-pull-request-templates/" rel="nofollow noopener">custom templates for issues</a>, and that is great. Tried the feature and it works perfectly. Now I am wondering if there’s a way to tune each template to perform specific queries automatically. </p>
<p>What I am specifically looking for is a way to encode the information to auto-label a new issues in the appropriate template. For instance if a user opens a bug report I want the tag “<em>bug</em>” to be automatically applied (given he has sufficient permissions to apply a label).</p>
<p>I have seen that I can do something like this using <a href="https://help.github.com/articles/about-automation-for-issues-and-pull-requests-with-query-parameters/" rel="nofollow noopener">queries</a> but the bit I am missing is if it is possible to generate those queries from the issue template.</p>
<p>Thanks in advance for any reply!</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/alalazo">@alalazo</a>,</p>
<p>This isn’t currently possible, but we’re always working to improve GitHub and consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by SaschaHeyer** [comment]
<p>Hey <a class="mention" href="/alalazo">@alalazo</a>,</p>
<p>have a look at <a href="https://defects.ai" rel="nofollow noopener">https://defects.ai<br>
</a>It is a service to predict labels for GitHub Issues fully automated, with machine learning models.<br>
I am the creator of Defects AI, reach out to me if you have any questions ✌️</p>
<p><img src="https://user-images.githubusercontent.com/1991664/181078742-4982760b-43b3-499c-90ca-cd56181e3a38.png" alt="labeling.png" data-base62-sha1="jZKztDazs72lsXSk8hQgTZuW2i4" width="532" height="195"></p>
<p>Cheers<br>
Sascha</p>

**Comment by gtskaushik** [comment]
<p>Started using the Issues Template. It is so cool.</p>
<p>Would be really happy to have labels attached along with templates.</p>

**Comment by jasonycw** [comment]
<p>I see the issue template now support labels, but not projects, will project tag be added?</p>
<p><a href="https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/" rel="nofollow noopener">https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2d1333eecd614985256409860947d802d18e3ea0.png" data-download-href="/uploads/short-url/6qKCWt2Phj9MnQXdB4PTwYBEXM4.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/4518597/181078745-522c0e50-7dd9-497c-a5ce-f355f589d753.png" alt="image.png" data-base62-sha1="6qKCWt2Phj9MnQXdB4PTwYBEXM4" width="690" height="287" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2d1333eecd614985256409860947d802d18e3ea0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">707×295 17.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/jasonycw">@jasonycw</a>,</p>
<p>We’re always working to improve GitHub, and we consider every suggestion we receive. I don’t know of any set plans to implement this, but I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by alalazo** [comment]
<p><a class="mention" href="/nadiajoyce">@nadiajoyce</a> Thank you for implementing the feature! One question: is there any way to apply multiple labels? For instance can I e.g. apply the labels “bug” and “help wanted” simultaneously to a type of report?</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/alalazo">@alalazo</a>,</p>
<p>Yes, indeed! You just need to list them, and separate with commas. Example: </p>
<p>labels: bug, feedback</p>
<p>^ That will produce an issue with the labels <code>bug</code> and <code>feedback</code> both applied.</p>
<p>Cheers!</p>

**Reply by bit-scientist** [reply]
I still didn't get it, sorry. Say, I have ISSUE_TEMPLATE in my .github. There are 4 `yaml` files for bug reports, docs, feature requests and questions. I want to automatically add an appropriate tag (for example: question tag for when a user creates an issue using the question template). What changes should be done on `question.yaml` file to achieve that?

**Comment by SueChaplain** [comment]
<p>Hi</p>
<p>I’m trying to add <code>title:</code> and <code>labels:</code> as YAML frontmatter but the feature just isn’t working for me. Can someone tell me what is wrong with the following please?:</p>
<hr>
<p>name: Release_A_Deliver<br>
title: “Deliver Release x.x”<br>
about: Epic for delivering next release<br>
labels: “jester:userdoc”, “jester:website”, “jester:other”</p>

**Comment by CathrineDreyer** [comment]
<p>Have to add to this thread too that adding labels automatically to a template would be very very useful! :smiley:</p>

**Comment by CathrineDreyer** [comment]
<p>Oh! It’s done! Perfect!</p>

**Comment by BardoBravo** [comment]
<p>Hi,</p>
<p>I am using a corporate account and the fields Title and Labels are filled. If I try in my private github account, it works fine.</p>
<p>Is there a limitation related to versioning?</p>
<p>Hugs,</p>
<p>Cláudio Ramos</p>

**Comment by bobily156** [comment]
<p>I have the same issue/need and we are using GitHub Enterprise 2.15.4</p>

**Comment by Ralf-Daimler** [comment]
<p>What about the projects … ? automatic assigning a project in an issue template</p>
<p>It does not work for me on <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a> </p>
<p>Please help … Thanks, Ralf</p>

**Comment by Puma-blip** [comment]
<blockquote>
<p><a class="mention" href="/jasonycw">@jasonycw</a> wrote:</p>
</blockquote>
<blockquote>
<p>I see the issue template now support labels, but not projects, will project tag be added?</p>
</blockquote>
<blockquote>
<p><a href="https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/" rel="nofollow noopener">https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/</a></p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p></p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2d1333eecd614985256409860947d802d18e3ea0.png" data-download-href="/uploads/short-url/6qKCWt2Phj9MnQXdB4PTwYBEXM4.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/63961224/181078754-bd7bd6e1-519f-445b-9b83-73a555528680.png" alt="image.png" data-base62-sha1="6qKCWt2Phj9MnQXdB4PTwYBEXM4" width="690" height="287" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2d1333eecd614985256409860947d802d18e3ea0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">707×295 17.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</blockquote>

**Comment by sidedwards** [comment]
<p>Fyi, for labels with spaces or characters that are not working with markdown or yaml formats you just need to wrap the value in quotes.</p>
<pre><code class="lang-auto">labels: "jester:userdoc, jester:website, jester:other"
</code></pre>

**Comment by mofosyne** [comment]
Also if possible I would love to have enumerated labels, so i can consolidate all these multiple issue template and have just a single dropdown label to select bug severity.

https://github.com/Mozilla-Ocho/llamafile/issues/new/choose
